src/neo/IO/Caching/HashCache.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+
+namespace Neo.IO.Caching
+{
+    public class HashCache<T> where T : IEquatable<T>
```

Comment:
Why not https://github.com/Qiao-Jin/neo/blob/da3b82a9ae958689f49847cc2a89d4eabc22695e/src/neo/IO/Caching/HashSetCache.cs ?

Reply:
The logic is somehow different, i.e. don't need a `bucketCapacity`, don't want to check `Contains` upon adding as has been checked before in `Blockchain.OnTransaction`, etc.